hail/src/main/scala/is/hail/io/fs/FS.scala

```diff
@@ -41,6 +41,16 @@ trait FileStatus {
   def getOwner: String
 }
 
+class BlobStorageFileStatus(path: String, modificationTime: java.lang.Long, size: Long, isDir: Boolean) extends FileStatus {
```

Comment:
If you look at the `GoogleStorageFileStatus` *object*, it uses an apply method to extract these values from the google object. We can do that too with `BlobProperties` if we create a similar `AzureStorageFileStatus`.

Reply:
I don't think we can exactly mimic that behavior since I don't believe that a directory has a BlobProperties object in the Azure API, but I've implemented something similar